pkg/proxy/envvar.go

```diff
@@ -0,0 +1,88 @@
+package proxy
+
+import (
+	"os"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// HTTP_PROXY is the URL of the proxy for HTTP requests.
+	envHTTPProxyName = "HTTP_PROXY"
+
+	// HTTPS_PROXY is the URL of the proxy for HTTPS requests.
+	envHTTPSProxyName = "HTTPS_PROXY"
+
+	// NO_PROXY is the list of domains for which the proxy should not be used.
+	envNoProxyName = "NO_PROXY"
+)
+
+var (
+	allProxyEnvVarNames = []string{
+		envHTTPProxyName,
+		envHTTPSProxyName,
+		envNoProxyName,
+	}
+)
+
+// SetEnvVars accepts a Proxy object and sets the proxy variables
+// as environment variables.
+func SetEnvVars(proxy *apiconfigv1.Proxy) {
+	if proxy == nil {
+		return
+	}
+	os.Setenv(envHTTPProxyName, proxy.Spec.HTTPProxy)
+	os.Setenv(envHTTPSProxyName, proxy.Spec.HTTPSProxy)
+	os.Setenv(envNoProxyName, proxy.Spec.NoProxy)
+}
+
+// GetPosition returns the index of the EnvVar with the given name.
+// If the list does not contain an EnvVar with the given name, -1 is returned.
+func GetPosition(list []corev1.EnvVar, name string) (position int, found bool) {
+	for i := range list {
+		if name == list[i].Name {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
+// Remove removes the given index from the list
+func Remove(list []corev1.EnvVar, index int) []corev1.EnvVar {
+	return append(list[:index], list[index+1:]...)
+}
+
+// Manage will ensure that the list of EnvVars are in sync with the proxy variables.
+// If the proxy variables are not set but are present in the list, they will be removed from the list.
+// If the proxy variables are set but missing from the list, they will be added to the list.
+// If the proxy variables are set and present in the list, their values will be updated in the list.
+func Manage(list []corev1.EnvVar) []corev1.EnvVar {
```

Comment:
I understand this was introduced so that you can ensure that you don't remove any user added environment variables plus you will only add the proxy variables only when required.. I am not sure if we want the user to be able to add their own environment variables given the `operator-registry` application is what is run by the deployment. Also given the list management you are doing here it might be preferable to always have the proxy environment variables in the spec even when they are empty.

In other words this could just be `SetEnvVars()`